scripts/path_plans.py

```diff
@@ -1,250 +1,140 @@
+        """
+        return self.path_as_transforms
 
-    def _setup_rings(self):
+    def get_path_as_poses(self):
+        # type: () -> List[List[float]]
+        """
+        Explicitly gets the path as a list of poses
+        @return: path as a list of poses [x, y, z, qw, qy, qz, qw]
+        """
+        return self.path_as_poses
 
-        # Generation
-        path_result = []
-        for lvl in self._levels[1:]:
+    def get_path_as_messages(self):
+        # type: () -> List[Dict]
+        """
+        Explicitly gets the path as a list of ROS pose messages
+        @return: path as a list of pose message {'Point': [x, y, z], 'Quaternion': [qx, qy, qz, qw]}
+        """
+        return self.path_as_messages
 
-            # Create Ring
-            pheta = np.linspace(-np.pi, np.pi, self._density, endpoint=False)
-            xx = (self._min_diameter * np.cos(pheta))
-            yy = (self._min_diameter * np.sin(pheta))
-            zz = (np.full_like(pheta, lvl))
+    def _setup_path(self):
+        # type: () -> List[np.ndarray]
+        """
+        Creates a list of path transformations that position the camera around
+        the part and orient the Z-axis towards the part's center location with the
+        Y-axis parallel to the XY plane and X-axis pointing up.
 
-            # Create Transforms
-            for x,y,z in zip(xx,yy,zz):
+        Usage: automatically called by __init__
 
-                rot_matrix = self._rotationMatrix_orient_z_axis([x,y,z])
-                path_result.append( self.tf.generateTransMatrix(rot_matrix,[x,y,z]) )
+        @return: The path as a list of transformation matrices
+        """
+        ''' OVERWRITE THIS FUNCTION IN YOUR PATH PLAN '''
+        path_as_transforms = []
+        return path_as_transforms
 
-        # Shift path body to be framed in context of Global Fixed Frame (0,0,0)
-        self._path_tf = self.tf.convertPath2FixedFrame(path_result, self._locator)
+    @staticmethod
+    def _convert_poses_to_messages(path_poses):
+        # type: (List[List[float]]) -> List[Dict]
+        """ Takes a path as a list of poses and formats them into a
+        dictionary per: http://docs.ros.org/en/jade/api/geometry_msgs/html/msg/Pose.html
 
-        # Convert path to Robot Poses (outputs a list of vectors x,y,z,qx,qy,qz,qw)
-        self._path_pose = self.tf.convertPath2RobotPose(self._path_tf)
-        
-        return
+        Each element in the returned list looks like:
+        msg = {'Point': [x, y, z], 'Quaternion': [x, y, z, w]}
 
-    def _rotationMatrix_orient_z_axis(self, local_point):
+        @param path_poses: List of poses as [x, y, z, qx, qy, qz, qw]
+        @return: list of poses formatted as a dictionary
+        """
+        path_messages = []
+        for pose in path_poses:
+            msg = {
+                'Point': pose[:3],
+                'Quaternion': pose[3:]
+            }
+            path_messages.append(msg.copy())
+
+        return path_messages
```

Comment:
Just caught this when testing in ROS code. You cannot  assemble a ROS message like this. 
You need to build ROS messages by importing the object.

Reply:
I now realize this wasn't supposed to be a ROS message, but an internal formatting thing we were doing previously.
Let's go directly to a ROS message. I'm working on a commit that will do this.